build/xtask/src/config.rs

```diff
@@ -719,3 +665,72 @@ impl BuildConfig<'_> {
         cmd
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////
+
+fn read_and_flatten_toml(
+    cfg: &Path,
+    hasher: &mut DefaultHasher,
+    seen: &mut BTreeSet<PathBuf>,
+) -> Result<toml_edit::Document> {
+    use toml_patch::merge_toml_documents;
+
+    // Prevent diamond inheritance
+    if !seen.insert(cfg.to_owned()) {
+        bail!("{cfg:?} is inherited more than once.  Don't do that.");
```

Comment:
this is an amusing error but runs afoul of the "one sentence all lower case no periods" rule of thumb for error messages. would keep it to just the first sentence.

Reply:
Clarified in 5fc61b05e6a1f289db4862c7fcf603de62805ce3

I decided to explicitly say "diamond dependencies are not allowed" in the error messages, since just saying "X is inherited more than once" doesn't tell you why that's bad 🙃